tests/components/automation/test_init.py

```diff
@@ -572,3 +574,48 @@ def test_reload_config_handles_load_fails(self):
         self.hass.bus.fire('test_event')
         self.hass.block_till_done()
         assert len(self.calls) == 2
+
+
+    @asyncio.coroutine
+    def test_restore_state2(self):
+        """Ensure states are restored on startup."""
+        mock_restore_cache(self.hass, (
+            State('automation.hello', STATE_ON),
+            State('automation.bye', STATE_OFF),
+        ))
+
+        config = {automation.DOMAIN: [{
+            'alias': 'hello',
+            'trigger': {
+                'platform': 'event',
+                'event_type': 'test_event_hello',
+            },
+            'action': {'service': 'test.automation'}
+        }, {
+            'alias': 'bye',
+            'trigger': {
+                'platform': 'event',
+                'event_type': 'test_event_bye',
+            },
+            'action': {'service': 'test.automation'}
+        }]}
+
+        assert (yield from async_setup_component(
+            self.hass, automation.DOMAIN, config))
+
+        state = self.hass.states.get('automation.hello')
+        assert state
+        assert state.state == STATE_ON
+
+        state = self.hass.states.get('automation.bye')
+        assert state
+        assert state.state == STATE_OFF
+
+        self.hass.bus.fire('test_event_bye')
+        self.hass.block_till_done()
+        assert len(self.calls) == 0
+
+        self.hass.bus.fire('test_event_hello')
+        self.hass.block_till_done()
+        assert len(self.calls) == 1
+        assert len(self.calls) == 0  # why?
```

Comment:
I see also strange thing with assert in last time... I don't trust assert anymore